orca-api/src/main/java/com/netflix/spinnaker/orca/api/preconfigured/jobs/PreconfiguredJobConfigurationProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.api.preconfigured.jobs;
+
+import com.netflix.spinnaker.kork.annotations.Beta;
+import java.util.List;
+import org.pf4j.ExtensionPoint;
+
+/** Extension point for provisioning preconfigured Job stages. */
+@Beta
+public interface PreconfiguredJobConfigurationProvider extends ExtensionPoint {
```

Comment:
Any reason not to have `PreconfiguredJobStageProperties` be the `ExtensionPoint`?  Then in the `JobService` or whatever we may just be able to do:

```
  private JobConfigurationProperties jobConfigurationProperties;
  private Optional<List<PreconfiguredJobStageProperties>> preConfiguredJobStageProperties;

  @Autowired
  JobService(JobConfigurationProperties jobConfigurationProperties,
             Optional<List<PreconfiguredJobStageProperties>> preConfiguredJobStageProperties
  ) {
    this.jobConfigurationProperties = jobConfigurationProperties;
    this.preConfiguredJobStageProperties = preConfiguredJobStageProperties;
  }
```

I wonder if that would work?  If so, it feels like a cleaner way of adding an extension point rather than building the extension point around the `PreconfiguredJobConfigurationProvider` concept.

Reply:
I struggled with that too, but making a value object as an extension point somewhat looked odd because we wanted a builder/factory to actually build those config objects.